assets/js/components/surveys/SurveyViewTrigger.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * SurveyViewTrigger component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useMount } from 'react-use';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+
+const { useSelect, useDispatch } = Data;
+
+const SurveyViewTrigger = ( { triggerID, ttl } ) => {
+	const usingProxy = useSelect( ( select ) => select( CORE_SITE ).isUsingProxy() );
+
+	const { triggerSurvey } = useDispatch( CORE_USER );
+
+	useMount( () => {
+		if ( usingProxy ) {
+			triggerSurvey( { triggerID, options: { ttl } } );
```

Comment:
@danielgent this is still incorrect. Please, check the signature of the `triggerSurvey` action and update this call: https://github.com/google/site-kit-wp/blob/cfb5225d711d1eb6f812ca862f1494be91ba7566/assets/js/googlesitekit/datastore/user/surveys.js#L75-L85